metrics/metrics.go

```diff
@@ -0,0 +1,32 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/metrics/prometheus"
+)
+
+// MetricsServer runs and controls a HTTP pprof interface.
+type MetricsServer struct {
+	server *http.Server
+}
+
+func NewMetricsServer(port int, r metrics.Registry) *MetricsServer {
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", prometheus.Handler(r))
+	p := MetricsServer{
+		server: &http.Server{
+			Addr:    fmt.Sprintf(":%d", port),
+			Handler: mux,
+		},
+	}
+	return &p
+}
+
+// Listen starts the HTTP server in the background.
+func (p *MetricsServer) Listen() {
+	log.Info("metrics server stopped", "err", p.server.ListenAndServe())
```

Comment:
Shouldn't this be:
```suggestion
	log.Info("metrics server started", "err", p.server.ListenAndServe())
```

Reply:
It can be both. It can error immediately because, for example, the port is already used, it can error due to a signal after serving HTTP for a while and so on. I would keep `stopped` as it's more common or rephrase it completely.

Reply:
But.... it's already using `stopped`... I don't get what you want me to do here.

Reply:
Pedro proposed `started` instead. I don't want you to do anything, IMO it's good as it is. I think no one has strong opinion about this so it's good to merge.

Reply:
Ah, you're right, I'm blind I guess, didn't even notice Pedro there.